docs/user/build-customization.rst

```diff
@@ -104,6 +105,22 @@ To avoid this, it's possible to unshallow the :program:`git clone`:
        post_checkout:
          - git fetch --unshallow || true
 
+If your build also relies on the contents of other branches, it may also be necessary to re-configure git to fetch these:
+
+.. code-block:: yaml
+   :caption: .readthedocs.yaml
+
+   version: 2
+   build:
+     os: "ubuntu-20.04"
+     tools:
+       python: "3.10"
+     jobs:
+       post_checkout:
+         - git fetch --unshallow || true
+         - git config remote.origin.fetch '+refs/heads/*:refs/remotes/origin/*' || true
```

Comment:
Why is this line required? Seems pretty specific to a use case, and something I'd expect git to do by default?

Reply:
This is necessary because we're configuring git to only fetch a single branch in this command (which you'll see a variant of in all builds):

```
git fetch origin --force --prune --prune-tags --depth 50 refs/heads/main:refs/remotes/origin/main
```

If you shallow clone a repo then run the above locally and look at `.git/config`, you'll see something like the following:

```
[remote "origin"]
fetch = +refs/heads/main:refs/remotes/origin/main
```

Which means "map the `main` ref locally to the `main` remote ref". Running `git fetch` will always result in just this branch being fetched. To fix this, you need to tell git to fetch all branches and map them to the same name locally, which is what we're doing here.

Reply:
Oh, and as I mentioned in the blurb beforehand, you'll need to do this if you have a tool that needs access to other branches rather than just other tags.